sourcefiles/logicfactory.py

```diff
@@ -1784,3 +1805,43 @@ def getGameConfig(settings: rset.Settings, config: cfg.RandoConfig):
 
     return CfgType(settings, config)
 # end getGameConfig
+
+
+def get_rocksanity_location_groups():
+    denadoroRock = LocationGroup(
+        "Denadoro Rock", 1, lambda game: True, lambda weight: weight-1
+    )
+    denadoroRock.addLocation(Location(TID.DENADORO_ROCK))
+    yield denadoroRock
+
+    giantsClawRock = LocationGroup(
+        "Giantsclaw Rock", 1, lambda game: game.canAccessGiantsClaw()
+    )
+    giantsClawRock.addLocation(Location(TID.GIANTS_CLAW_ROCK))
+    yield giantsClawRock
+
+    # laruba
+    larubaRock = LocationGroup(
+        "Laruba Rock", 1, lambda game: game.canAccessPrehistory()
+    )
+    larubaRock.addLocation(Location(TID.LARUBA_ROCK))
+    yield larubaRock
+
+    # kajar
+    kajarRock = LocationGroup(
+        "Kajar Rock", 1, lambda game: game.canAccessMtWoe()
+    )
+    kajarRock.addLocation(Location(TID.KAJAR_ROCK))
+    yield kajarRock
+
+    # black omen
+    blackOmenRock = LocationGroup(
+        "Black Omen Rock", 1, lambda game: (
+            game.hasKeyItem(ItemID.PENDANT) and
+            game.hasKeyItem(ItemID.CLONE) and
+            game.hasKeyItem(ItemID.C_TRIGGER) and
+            (ItemID.JETSOFTIME if rset.GameFlags.EPOCH_FAIL else True)
```

Comment:
Lats comment was on an old version.  But this should include a hasKeyItem I believe.